doc: Added documentation about concurrent GPS and LTE in asset tracker

umapraseeda opened the pull request:
Added documentation about concurrent GPS and LTE in asset tracker
application documentation.
ref: https://projecttools.nordicsemi.no/jira/browse/NCSDK-4527

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>
[nRF9160_ Asset Tracker — nRF Connect SDK 1.2.99 documentation.pdf](https://github.com/NordicPlayground/fw-nrfconnect-nrf/files/4338489/nRF9160_.Asset.Tracker.nRF.Connect.SDK.1.2.99.documentation.pdf)

ru-fu commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,49 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+Concurrent GPS and LTE
+++++++++++++++++++++++
+
+An nRF9160-based device supports LTE-M and NB-IoT along with power saving features such as eDRX and PSM.
```
Spell out eDRX and PSM once?

ru-fu commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,49 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+	In this mode, the device maintains a continuous cellular link.
+	To enable this mode, set ``CONFIG_POWER_OPTIMIZATION_ENABLE=n``.
+
+Request eDRX mode
```
Spell out eDRX once?

umapraseeda replied:
Added.

umapraseeda commented:
@jtguggedal

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,49 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+==================
+
+The Asset Tracker can run in three power modes that are configured in the Kconfig file of the application.
+These settings are currently only supported on the nRF9160 DK.
```
The modes can be used also on Thingy:91, but they can't be changed runtime using the switches

tokangas replied:
It would be good to mention the configurations used to enable eDRX and PSM as well.

umapraseeda replied:
@jtguggedal Added the sentence: Although these power modes are supported in Thingy:91, the settings cannot be changed at runtime.

@tokangas  The configurations are mentioned in the Power Optimization section.

tokangas replied:
I think this is a bit confusing, because it first says that these settings are only supported on the DK and then continues saying that the modes are supported in Thingy:91, but can't be changed at runtime.

The problem is that there are two different things here:

- changing between eDRX and PSM during runtime (supported only by DK)
- build time configuration for enabling eDRX and/or PSM (supported by both)

The first one can be enabled by setting CONFIG_POWER_OPTIMIZATION_ENABLE and then the used power power saving mode (eDRX/PSM) can be selected using the HW switch as described by the text currently.

However, to enable eDRX and/or PSM for Thingy:91 (this of course also works with DK as well), different configurations are used. eDRX can be enabled by setting CONFIG_LTE_EDRX_REQ and for PSM there's CONFIG_GPS_CONTROL_PSM_ENABLE_ON_START, which enables PSM when the GPS starts. CONFIG_POWER_OPTIMIZATION_ENABLE does nothing on a Thingy:91.

I think we should either change this to clearly describe the runtime power optimization mode change and say that it's only supported by DK or cover both possibilities separately.

JaniM74 replied:
Further comment to earlier chapter:
GPS coordinates | GPS | NEMA Gxxx string ---> **NMEA** Gxxx string

umapraseeda replied:
Added the suggested information.

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,49 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+
+Demo mode
+	This is the default setting.
+	In this mode, the device maintains a continuous cellular link.
```
To be fair to eDRX and PSM, they also offer a continuous link, in the sense that the connection to the cellular network is maintained.
Without eDRX and PSM, the device can receive data at all times, while it cannot with those two modes enabled. 
Then it's only reachable at certain configurable intervals, or when the device sends data to the network.

umapraseeda replied:
Added the above information.

jtguggedal commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,49 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+++++++++++++++++++++++
+
+An nRF9160-based device supports LTE-M and NB-IoT along with power saving features such as eDRX and PSM.
+The following figure shows how the data transfer occurs in an nRF9160-based device with energy conservation in place.
```
GPS will also work without these two modes, but it's very case-dependant if that works well. Generally, we would very much encourage the use of eDRX and/or PSM (both can be used at the same time) when using the GPS.

umapraseeda replied:
Added the information:  Concurrent operation of GPS with optional power saving features such as extended Discontinuous Reception (eDRX) and Power Saving Mode (PSM) is also supported and recommended.

jhn-nordic commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,49 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+The time between the transition of a device from RRC connected mode (data transfer mode) to RRC idle mode is dependent on the network.
+Typically the time ranges between 5 seconds to 70 seconds after the last data transfer on LTE.
+
+A GPS operation occurs when the device is either not reachable (PSM) or when the device is in the sleep mode (eDRX).
```
This is not completely correct. We state above "Asset tracker enables the working of concurrent GPS and LTE in eDRX and PSM modes when the device is in `RRC idle mode`_."
I think this linecan be removed." GPS operation occurs when the device is either not reachable (PSM) or when the device is in the sleep mode (eDRX)."

umapraseeda replied:
Removed the sentence.

jtguggedal commented:
Thinking a bit more about this, shuld we have a separate sectionin the docs somewhere that covers eDRX and PSM in more deatil that we could link to in the various samples/apps? The settings are relevant for all things LTE, and apply to pretty much all nRF91 samples/apps.

jtguggedal commented:
@tokangas Could you have a look at this as well?

tokangas commented:
I think it's a good idea to add a separate section for describing eDRX and PSM. @JaniM74 should at least check it, maybe even help write it.

umapraseeda commented:
> I think it's a good idea to add a separate section for describing eDRX and PSM. @JaniM74 should at least check it, maybe even help write it.
Moved this section to 'Working with nRF9160' user guide and added the same information in the 'Working with Thingy:91' user guide also.

umapraseeda commented:
> I think it's a good idea to add a separate section for describing eDRX and PSM. @JaniM74 should at least check it, maybe even help write it.

Moved this section to 'Working with nRF9160' user guide and added the same information in the 'Working with Thingy:91' user guide also.

JaniM74 commented on applications/asset_tracker/README.rst:
```diff
@@ -52,6 +52,36 @@ The `LTE Link Monitor`_ application, implemented as part of `nRF Connect for Des
+
+.. note::
+   Not all cellular network providers support these modes, and the granted parameters can vary between networks.
+
```
Could also state that SIM card subscription also effects the availability of the Cellular IoT power saving features. Network operators can also limit the availability of power saving features for Roaming users

mia-ko commented on applications/asset_tracker/README.rst:
```diff
@@ -53,6 +53,39 @@ You can also send AT commands from the **Terminal** card on nRF Cloud when the d
+
+The Asset Tracker can run in three power modes that are configured in the Kconfig file of the application.
+Currently the dynamic switching between eDRX and PSM during run-time is only supported on the nRF9160 DK.
+However, both nRF9160 DK and Thingy:91 supports build-time configuration for enabling eDRX, PSM or both.
```
supports -> support

mia-ko commented on applications/asset_tracker/README.rst:
```diff
@@ -53,6 +53,39 @@ You can also send AT commands from the **Terminal** card on nRF Cloud when the d
+
+Request eDRX mode
+	In this mode, the device requests the eDRX feature from the cellular network to save power.
+	In this mode, the device maintains a continuous cellular link.
```
In this mode, the device maintains... -> The device maintains...
(Two consecutive sentences starting with "In this mode")

mia-ko commented on applications/asset_tracker/README.rst:
```diff
@@ -53,6 +53,39 @@ You can also send AT commands from the **Terminal** card on nRF Cloud when the d
+
+Request Power Saving Mode (PSM)
+	In this mode, the device requests the PSM feature from the cellular network to save power.
+	In this mode, the device maintains a continuous cellular link.
```
Two consecutive sentences starting with "In this mode"

mia-ko commented on doc/nrf/ug_nrf9160.rst:
```diff
@@ -162,6 +162,31 @@ If the modem is shut down gracefully before the next boot (by using ``AT+CFUN=0`
+======================
+
+|An nRF9160-based device| supports GPS in LTE-M and NB-IoT.
+Concurrent operation of GPS with optional power saving features such as extended Discontinuous Reception (eDRX) and Power Saving Mode (PSM) is also supported and recommended.
```
Add comma:
Concurrent operation of GPS with optional power saving features, such as extended Discontinuous Reception (eDRX) and Power Saving Mode (PSM), is also supported and recommended.

mia-ko commented on doc/nrf/ug_nrf9160.rst:
```diff
@@ -162,6 +162,31 @@ If the modem is shut down gracefully before the next boot (by using ``AT+CFUN=0`
+|An nRF9160-based device| supports GPS in LTE-M and NB-IoT.
+Concurrent operation of GPS with optional power saving features such as extended Discontinuous Reception (eDRX) and Power Saving Mode (PSM) is also supported and recommended.
+
+The following figure shows how the data transfer occurs in |an nRF9160-based device| with energy conservation in place.
```
Energy conservation...
Not sure if you need to change this... I was just curious why one wouldn't use power saving instead.

ru-fu commented:
Guess this one is ready to merge now?

umapraseeda commented:
> Guess this one is ready to merge now?

Yes.
